### PR TITLE
Check for maximum scale in IOBuffer before array copy

### PR DESCRIPTION
**Pull request by Jeffery-Wasty**
Input was not validated for max length before being copied to byte array.

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java [hunk 1]
```diff
@@ -449,42 +449,6 @@ public void testConnectionPoolGetTwice() throws SQLException {
         }
     }
 
-    /**
-     * Tests whether connectRetryCount and connectRetryInterval are properly respected in the login loop. As well, tests
-     * that connection is retried the proper number of times.
-     */
-    @Test
-    public void testConnectCountInLoginAndCorrectRetryCount() {
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java:457]: why was this test removed what does it have to do with what changed in this PR??

File: src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPTypesTest.java [hunk 2]
```diff
@@ -573,6 +574,34 @@ public String toString() {
...
+        tvp.addColumnMetadata("c1", java.sql.Types.NUMERIC);
+        tvp.addRow(new BigDecimal(0.222));
+
+        final String sql = "{call " + AbstractSQLGenerator.escapeIdentifier(procedureName) + "(?)}";
+
+        try (SQLServerCallableStatement callableStmt = (SQLServerCallableStatement) connection.prepareCall(sql)) {
+            callableStmt.setStructured(1, tvpName, tvp);
+            callableStmt.execute();
+
+            assertEquals(false, true, TestResource.getResource("R_expectedExceptionNotThrown"));
```
Comment (tkyc) [inline src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPTypesTest.java:598]: You can use `fail(TestResource.getResource....)`

**Review by tkyc — DISMISSED** [review]

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java [hunk 3]
```diff
@@ -540,7 +540,8 @@ protected Object[][] getContents() {
         {"R_serverError", "An error occurred during the current command (Done status {0}). {1}"},
         {"R_ManagedIdentityTokenAcquisitionFail", "Failed to acquire managed identity token. Request for the token succeeded, but no token was returned. The token is null."},
         {"R_AmbiguousRowUpdate", "Failed to execute updateRow(). The update is attempting an ambiguous update on tables \"{0}\" and \"{1}\". Ensure all columns being updated prior to the updateRow() call belong to the same table."},
-        {"R_InvalidSqlQuery", "Invalid SQL Query: {0}"}
+        {"R_InvalidSqlQuery", "Invalid SQL Query: {0}"},
+        {"R_InvalidScale", "Scale of input value is larger than the maximum allowed by SQL Server (38)."}
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java:544]: maybe we don't need to explicitly say 38 here? just max is enough

Comment (Jeffery-Wasty) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResource.java:544]: Sure, I can remove that.

**Review by tkyc — APPROVED** [review]

File: src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPTypesTest.java [hunk 4]
```diff
@@ -573,6 +574,34 @@ public String toString() {
...
+
+        final String sql = "{call " + AbstractSQLGenerator.escapeIdentifier(procedureName) + "(?)}";
+
+        try (SQLServerCallableStatement callableStmt = (SQLServerCallableStatement) connection.prepareCall(sql)) {
+            callableStmt.setStructured(1, tvpName, tvp);
+            callableStmt.execute();
+
+            fail(TestResource.getResource("R_expectedExceptionNotThrown"));
+        } catch (IllegalArgumentException e) {
+            assertEquals("Scale of input value is larger than the maximum allowed by SQL Server.",
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPTypesTest.java:600]: don't hardcode, use formatErrorMsg to match, see other tests

**Review by lilgreenbird — APPROVED** [review]

**Review by tkyc — APPROVED** [review]

**Pull request merged**